contracts/bonding/BondingManager.sol

```diff
@@ -249,43 +238,39 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             }
         }
 
-        if (del.delegateAddress == address(0)) {
-            // If no existing delegate, delegation amount = bonded stake + amount included in this function call
-            delegationAmount = delegationAmount.add(del.bondedAmount);
-        }
-
+        // Delegation amount must be > 0 - cannot delegate to someone without having bonded stake
+        require(delegationAmount > 0);
```

Comment:
I don't think there is any reason for someone to have a delegate unless they also have bonded stake, so this check makes sure the delegation amount (sum of current bonded stake and provided amount) is greater than 0